src/mesi_search/candig.py

```diff
@@ -62,7 +62,11 @@ def data_filter(data={}, terms=[], path=""):
             if patients_data and len(patients_data) == 1:  # TODO: check for `>1`
                 term_results = {}
                 for term in terms:
-                    term_results[term] = dpath.util.get(patients_data[0], term)
+                    try:
+                        term_results[term] = dpath.util.get(patients_data[0], term)
+                    except KeyError as k:
+                        logger.error("Did not find the term {} "
+                                     "in dataset {}".format(term, dataset_id), str(k))
```

Comment:
Looking at the docs for `dpath.util.get` it seems like `ValueError` can be raised as well: `If more than one leaf matches the glob, ValueError is raised.` I'm not sure if you need to account for this error in your implementation but if so, you should add an except statement to catch ValueError.

Reply:
This is an excellent observation! My belief was that there is only one list item for `/results/patients` that we gather by `patients_data[0]`. This single item cannot have a duplicate dictionary key because it is not allowed in python dicts. So, each `term` is unique within the `patients_data[0]` object. 

What do you think?

Reply:
If each `term` is unique in `patients_data[0]` then I don't think you need to add a catch statement for `ValueError`. Now thinking about it, I think that error happens when you pass in a regex expression to get (something like `/x/*`. This regex may evaluate to more than one key so then a ValueError would be raised.)